addons/transparent-orphans/userscript.js

```diff
@@ -0,0 +1,80 @@
+export default async function ({ addon, console }) {
+  let blockly = await addon.tab.traps.getBlockly();
+
+  let transparentBlocks = [];
+
+  function setTransparency(block, t) {
+    let current = block;
+
+    current.setOpacity(t);
+
+    for (let child of current.getChildren()) {
+      setTransparency(child, t);
+    }
+
+    for (let input of current.inputList) {
+      for (let dropdown of input.fieldRow.filter((d) => d.argType_ && d.box_)) {
+        dropdown.box_.style.fillOpacity = t;
+      }
+
+      if (input.type === 1 && input.outlinePath) {
+        input.outlinePath.style.fillOpacity = t;
+      }
+    }
+
+    while (current.getNextBlock()) {
+      current = current.getNextBlock();
+
+      current.setOpacity(t);
+
+      for (let child of current.getChildren()) {
+        setTransparency(child, t);
+      }
+
+      for (let input of current.inputList) {
+        for (let dropdown of input.fieldRow.filter((d) => d.argType_)) {
+          dropdown.box_.style.fillOpacity = t;
+        }
+
+        if (input.type === 1) {
+          input.outlinePath.style.fillOpacity = t;
+        }
+      }
+    }
```

Comment:
This duplicates lines 9-23. Is it possible to move it out into its own function?

Reply:
Maybe start current as `block.getParent()`? But that would not work for reporters, it would just skip them...